src/js/components/Menu/Menu.js

```diff
@@ -367,25 +367,30 @@ const Menu = forwardRef((props, ref) => {
       <Box
         // eslint-disable-next-line react/no-array-index-key
         key={groupIndex}
-        {...theme.menu.group?.container}
-        border={
-          groupIndex > 0
-            ? {
-                side: 'top',
-                color: theme.menu.group?.separator?.color,
-                size: theme.menu.group?.separator?.size,
-              }
-            : undefined
-        }
       >
-        {group.map((item) => {
-          // item index needs to be its index in the entire menu as if
-          // it were a flat array
-          const currentIndex = index;
-          index += 1;
-
-          return menuItem(item, currentIndex);
-        })}
+        <Box pad={theme.menu.group.separator.pad}>
+          <Box
```

Comment:
Are we missing tests for a grouped menu? I would've expected snapshots to update

Reply:
We had a test for grouped menu but the menu items weren't showing in the snapshot. I changed the test so that the menu items are showing.